content/articles/2019-08-lifecycle-0-1-0.Rmarkdown

```diff
@@ -0,0 +1,255 @@
+install.packages("lifecycle")
+```
+
+In this post, we'll briefly present the lifecycle workflow. Read the [Get started](http://lifecycle.r-lib.org/articles/lifecycle.html) vignette for a more complete description. Normally you'd call lifecycle functions with the namespace prefix, but for the purposes of illustrating the package we're attaching the package with `library()`:
+
+```{r, message = FALSE}
+library(lifecycle)
+```
+
+
+## Lifecycle stages
+
+The stages for functions and arguments are modelled after the [lifecycle stages for packages](https://www.tidyverse.org/lifecycle/).  There are 4 __development__ stages. Then, if the decision of discontinuing a feature has been made, it goes through 3 __deprecation__ stages.
+
+### Development stages
+
+```{r echo = FALSE}
+knitr::include_graphics("/images/lifecycle/lifecycle-experimental.svg")
+```
+
+An __experimental__ feature is in the very early stage of development. It is exported so users can start to use it and report feedback, but its interface and/or behaviour is likely to change in the future. It is generally best to avoid depending on experimental features.
+
+```{r echo = FALSE}
+knitr::include_graphics("/images/lifecycle/lifecycle-maturing.svg")
+```
+
+The interface and behaviour of a __maturing__ feature has been roughed out, but finer details are likely to change. It still needs more feedback to find the optimal API.
+
+```{r echo = FALSE}
+knitr::include_graphics("/images/lifecycle/lifecycle-stable.svg")
+```
+
+A feature is considered __stable__ when the author is happy with its interface and behaviour. Major changes are unlikely, and breaking changes will occur gradually, through a deprecation process.
+
+```{r echo = FALSE}
+knitr::include_graphics("/images/lifecycle/lifecycle-questioning.svg")
+```
+
+In the __questioning__ stage, the author is no longer convinced that the feature is the optimal approach. However, there are no recommended alternatives yet.
+
+
+### Deprecation stages
+
+When the author is no longer happy with a feature because they consider it sub-optimal compared to some other approach, or simply because they no longer have the time to maintain it, it goes through three stages of deprecation.
+
+```{r echo = FALSE}
+knitr::include_graphics("/images/lifecycle/lifecycle-soft-deprecated.svg")
+```
+
+A __soft-deprecated__ feature can still be used without hassle, but users should consider switching to an alternative approach.
+
+```{r echo = FALSE}
+knitr::include_graphics("/images/lifecycle/lifecycle-deprecated.svg")
+```
+
+A __deprecated__ feature is likely to be discontinued in the next major release. Users should switch to an alternative approach as soon as possible.
+
+```{r echo = FALSE}
+knitr::include_graphics("/images/lifecycle/lifecycle-defunct.svg")
+```
+
+A __defunct__ function is still exported, and a defunct argument is still part of the signature. However, the feature can no longer be used, it now throws an informative error.
```

Comment:
(In the tidyverse, we try to keep defunct functions around for a long, so that when you're using old code with new packages, you still get pointed in the right direction for updating)